### PR TITLE
[MIG][10.0] bi_sql_editor

### PR DESCRIPTION
**Pull request by p-tombez**
Take over of #162

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why are you opening this without asking in the other PR?

**Review by AaronHForgeFlow — COMMENTED** [review]
Thank you for the fixes.

Some comments below:

I'm having trouble to delete the materialized views. Cannot set to draft the bi report. Please check on runbot the sales by customer report (maybe it has something to do withthe sql field type, which is wrong ).

Also, IMHO we should not permit to duplicate the views, it is dangerous.

PS: I agree with @pedrobaeza  you should ask first in the other PR is they are going to work on that. If @darshan-serpent is not going to continue maintaining that PR, then you can continue with this PR.

**Comment by p-tombez** [comment]
@aheficent The problem with the materialized view should be solved now, can you test it ?
You say that duplicating views is dangerous, can you tell me more about what you think could be dangerous ? Maybe I can improve the ```copy``` method given your feedback

@pedrobaeza I tried to mail @darshan-serpent a week ago but didn't get any answer and he is not replying to the comments on the original PR since 3 months. Knowing that, and that we need this module to work ASAP, we decided to take over the PR

**Comment by pedrobaeza** [comment]
@p-tombez OK, thanks for the explanations. It's good to include all this information here in public for avoid misunderstandings.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Solved the issue with materialized views. The copy method is working fine. Please dismiss my previous comment, I did not like the idea of duplicating views, but it's really not dangerous. Thank you.

File: bi_sql_editor/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'BI SQL Editor',
+    'summary': 'BI Views builder, based on Materialized or Normal SQL Views',
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'category': 'Reporting',
+    'author': 'GRAP,Odoo Community Association (OCA)',
+    'website': 'https://www.odoo-community.org',
```
Comment (simahawk) [inline bi_sql_editor/__manifest__.py]: https://github.com/OCA/reporting-engine

**Review by simahawk — APPROVED** [review]
LGTM

**Review by leemannd — APPROVED** [review]
Code review and @p-tombez showed it to me

**Comment by CasVissers-360ERP** [comment]
I still git this error when doing an update all:
psycopg2.ProgrammingError: "x_bi_sql_view_kickback_fee" is not a table, composite type, or foreign table

**Comment by CasVissers-360ERP** [comment]
@p-tombez would it be possible to take fix the "is not a table, composite type, or foreign table"? It's quiet nasty if we want to use this module in a production environment.

**Comment by leemannd** [comment]
@CasVissers Hum, could you point out where is the fix or do a pull-request in order to speedup its inclusion?

**Comment by legalsylvain** [comment]
Hi @all,
I justed tested and reviewed the whole module and made some fixes and improvements you can see in this PR : https://github.com/p-tombez/reporting-engine/pull/1

Could you review and merge it. 
Thanks.

kind regards. 

Improvment
----
* is_materialized field is now not readonly on sql_valid state
* better display of the field group_ids in the view form
* Add possibility to reorder menu items from sql views, via new sequence field
* Do not warn user when setting sql view to draft if state is sq_valid

Fixes
---
* prevent possibility to set indexes on non materialized view. (generate an error)
* domain_force and group_ids are now readonly if state > sql_valid
* Set Date of the first execution in the action name if the view is materialized

I also made a trivial fix against server-tools. sql_request_abstract module, you can review.

https://github.com/OCA/server-tools/pull/1312

**Comment by legalsylvain** [comment]
@p-tombez : could you review and merge my PR ? 

kind regards.

**Comment by p-tombez** [comment]
@legalsylvain I reviewed it and merged it. That's some nice additions.
Can everyone take a look again and validate the new changes ?

**Comment by legalsylvain** [comment]
@p-tombez Thanks for the review and the merge !

@all : thanks for a last :+1: to confirm the last changes.

regards.

**Review by leemannd — APPROVED** [review]
@legalsylvain Thanks!

**Review by CasVissers-360ERP — APPROVED** [review]
Thnx everyone. I tested module multiple times. Time to merge!

**Comment by JordiBForgeFlow** [comment]
Merged now. Thanks to all contributors!!!

**Pull request merged**